control_plane/src/storage.rs

```diff
@@ -46,27 +44,36 @@ impl PageServerNode {
 
         PageServerNode {
             kill_on_exit: false,
-            pg_connection_config: Self::default_config(password), // default
+            pg_connection_config: Self::pageserver_connection_config(
+                password,
+                env.pageserver_pg_port,
+            ),
             env: env.clone(),
             http_client: Client::new(),
-            http_base_url: HTTP_BASE_URL.to_owned(),
+            http_base_url: format!("http://localhost:{}/v1", env.pageserver_http_port),
         }
     }
 
-    fn default_config(password: &str) -> Config {
-        format!("postgresql://no_user:{}@localhost:64000/no_db", password)
+    fn pageserver_connection_config(password: &str, port: u16) -> Config {
+        format!("postgresql://no_user:{}@localhost:{}/no_db", password, port)
             .parse()
             .unwrap()
     }
 
     pub fn init(&self, create_tenant: Option<&str>, enable_auth: bool) -> Result<()> {
         let mut cmd = Command::new(self.env.pageserver_bin()?);
+        let listen_pg = format!("localhost:{}", self.env.pageserver_pg_port);
+        let listen_http = format!("localhost:{}", self.env.pageserver_http_port);
         let mut args = vec![
             "--init",
             "-D",
             self.env.base_data_dir.to_str().unwrap(),
             "--postgres-distrib",
             self.env.pg_distrib_dir.to_str().unwrap(),
+            "--listen-pg",
+            &listen_pg,
+            "--listen-http",
+            &listen_http,
```

Comment:
Hm, this way pageserver will save this in a config file, probably reduntant, will remove

Reply:
It turns out that I've donee forwarding of arguments in zenith init, not in zenith start, so currently this worked because values were saved to config. IMO it is ok to set ports in config during init and use them during start. I'd rather leave it as is, what do you think?

Reply:
> so currently this worked because values were saved to config.

Could you clarify what worked when? Are you saying that omitting those args worked because they were saved to config, or something else?

Reply:
Other way around. It worked when variables were passed to `zenith init`. They were forwarded to `pageserver init` and saved to pageserver config. Then `zenit start` do not forward anything, and pageserver uses values from config. When I removed this forwarding pageserver falled back to defaults

Reply:
Basically this is about where to pass this args, to `zenith init` or to `zenith start`. It may be both, but currently only init forwards them to pageserver